sentinelhub/sentinelhub_batch.py

```diff
@@ -0,0 +1,398 @@
+        :type grid_id: str or int
+        :param config: A configuration object
+        :type config: SHConfig
+        :return: A tiling grid definition
+        :rtype: dict
+        """
+        url = f'{SentinelHubBatch._get_tiling_grids_url(config)}/{grid_id}'
+        return get_auth_json(url)
+
+    @property
+    def info(self):
+        """ A dictionary with a Batch request information. It loads a new dictionary only if one doesn't exist yet.
+
+        :return: Batch request info
+        :rtype: dict
+        """
+        if self._request_info is None:
+            self.update_info()
+        return self._request_info
+
+    def update_info(self):
+        """ Updates information about a batch request
+
+        :return: Batch request info
+        :rtype: dict
+        """
+        url = self._get_process_url(self.config, request_id=self.request_id)
+        self._request_info = get_auth_json(url)
+
+    @property
+    def evalscript(self):
+        """ Provides an evalscript used by a batch request
+
+        :return: An evalscript
+        :rtype: str
+        """
+        return self.info['processRequest']['evalscript']
+
+    @property
+    def bbox(self):
+        """Provides a bounding box used by a batch request
+
+        :return: An area bounding box together with CRS
+        :rtype: BBox
+        """
+        bbox, _, crs = self._parse_bounds_payload()
+        return BBox(bbox, crs)
+
+    @property
+    def geometry(self):
+        """ Provides a geometry used by a batch request
+
+        :return: An area geometry together with CRS
+        :rtype: Geometry
+        """
+        _, geometry, crs = self._parse_bounds_payload()
+        if geometry is None:
+            geometry = self.bbox.geometry
+        return Geometry(geometry, crs)
+
+    @staticmethod
+    def iter_requests(search=None, sort=None, user_id=None, config=None, **kwargs):
+        """ Iterate existing batch requests
+
+        :param search: Filter requests by a search query
+        :type search: str or None
+        :param sort: Sort obtained batch requests in a specific order
+        :type sort: str or None
+        :param user_id: Filter requests by a user id who defined a request
+        :type user_id: str or None
+        :param config: A configuration object
+        :type config: SHConfig or None
+        :param kwargs: Any additional parameters to include in a request query
+        :return: An iterator over existing batch requests
+        :rtype: Iterator[SentinelHubBatch]
+        """
+        url = SentinelHubBatch._get_process_url(config)
+        params = _remove_undefined_params({
+            'search': search,
+            'sort': sort,
+            'userid': user_id,
+            **kwargs
+        })
+        for request_info in _iter_pages(url, **params):
+            yield SentinelHubBatch(request_info=request_info, config=config)
+
+    @staticmethod
+    def get_latest_request(config=None):
```

Comment:
This will retrieve all requests from service and then only return the last one. Any chance the service could do that? Or that we'd create request using `sort` parameter and only request first page?

Reply:
Yes, in the future `sort` parameter will be used for this. At the moment it is not yet implemented even though it is written in documentation.

For now I don't see any other way how to convince the service to only give the last request. The service is returning requests in a seemingly arbitrary order.

Reply:
Ah, ok. Then let's keep it as is.